### DISCUSSION TITLE
A "Stop what you're doing" button

### DISCUSSION THREAD

**Discussion by batnom** [Beyond-Better/bb#30] [Ideas]
Sometimes I have seen BB go above and beyond with a request, sometimes this is helpful but other times it is not.

For example I ask for it to resolve some build errors after a large refactor, but then it starts adding brand new functionality to the game (i'm guessing as it resolved build errors).

The next time I even asked it to "do not touch any game mechanics or features, I just want you to fix these build errors: ..." but then it started adding brand new features into the game.

As this was at the end of a large refactor the conversation was quite long, so each time it did this it used up a lot of tokens.

![Screenshot 2024-11-12 at 11 01 30 am](https://github.com/user-attachments/assets/de2c7477-415d-4fa0-be71-bfd315893fdc)

^ I kept it going hoping it would eventually finish.  It ended up finishing with `{"entryType":"answer"}` and with build errors remaining :( 

Having a button to abort whatever it is doing would be good to stop this in its tracks, without having to do a manual `bb stop`.

**Comment by cngarrison** [comment]
The new BUI comes with a "Stop" button during the "Claude is working" status.
